src/export/StructureDefinitionExporter.ts

```diff
@@ -186,51 +172,48 @@ export class StructureDefinitionExporter {
    * @returns {StructureDefinition}
    * @throws {ParentNotDefinedError} when the Profile or Extension's parent is not found
    */
-  exportStructDef(fshDefinition: Profile | Extension): void {
-    if (this.structDefs.some(sd => sd.name === fshDefinition.name)) {
+  exportStructDef(fshDefinition: Profile | Extension): StructureDefinition {
+    if (
+      this.pkg.profiles.some(sd => sd.name === fshDefinition.name) ||
+      this.pkg.extensions.some(sd => sd.name === fshDefinition.name)
+    ) {
       return;
     }
 
     const parentName = fshDefinition.parent || 'Resource';
-    const structDef = this.resolve(parentName);
+    const json = this.fishForFHIR(parentName);
 
     // If we still don't have a resolution, then it's not defined
-    if (!structDef) {
+    if (!json) {
       throw new ParentNotDefinedError(fshDefinition.name, parentName, fshDefinition.sourceInfo);
     }
 
+    const structDef = StructureDefinition.fromJSON(json);
```

Comment:
Yeah, so it's essentially doing what you say above (re: Profile A and Profile B).  There's a few reasons for that.

(1) I wanted a consistent approach to the `fishForFHIR` function.  The `fishForFHIR` function can get a result from the FHIRDefinitions (which is straight JSON) or from the Package (which is an instance of StructureDefinition).  For consistency, regardless of the source, the returned type should be the same -- so I had to decide between instances of `StructureDefinition`, `ValueSet`, and `CodeSystem`; or plain JSON for everything.  The problem with using instances of the specific classes is that `ValueSet` and `CodeSystem` don't have `fromJSON` functions yet, so when we get one of those from FHIRDefinitions we don't have a way to transform it into an instance of its respective class.  I could have implemented that, but it seemed low value.  It's also possible that the caller really only needs the JSON anyway, so half the time it might be useless.  So I decided just to have `fishForFHIR` always return JSON -- and here we are.

This actually also unintentionally fixed a bug that I didn't realize we had.  Before, if we started w/ the parent's `StructureDefinition`, we weren't recapturing the original, so the child ended up thinking that changes that the parent made were actually its own changes and putting them in its own diff.  You can see that if you compare an export of fsh-mcode from master to an export from this branch.